src/infrastructure/GormDbInitializer.go

```diff
@@ -22,6 +26,55 @@ func (ns OurNamingSchema) TableName(str string) string {
 	return defaultNamingStrategy.TableName(str)
 }
 
+func createMySQLDb(cfg domain.MySQL) (*gorm.DB, error) {
+	connectionString := fmt.Sprintf("%s:%s@%s(%s:%s)/", cfg.Username, cfg.Password,
+		cfg.Protocol, cfg.Hostname, cfg.Port)
+	dialector := mysql.Open(fmt.Sprintf("%s%s%s", connectionString, cfg.DbName, cfg.Parameters))
+	db, err := gorm.Open(dialector, &gorm.Config{
+		Logger: logger.Default.LogMode(logger.Info),
+	})
+	if err != nil {
+		return nil, errors.Internal.Wrap(err, "failed to open mysql db")
+	}
+	err = createDbIfNotExists(db, cfg.DbName)
+	if err != nil {
+		return nil, errors.Internal.Wrap(err, "error creating db")
+	}
+	return db, nil
+}
+
+func createSQLiteDb(cfg domain.SQLite) (*gorm.DB, error) {
+	dbPath := cfg.Filename
+	pathSplit := strings.Split(dbPath, "/")
+	dbName := pathSplit[len(pathSplit)-1]
+
+	if dbPath[0:1] != "/" {
+		executablePath, _ := os.Executable()
+		dbPath = path.Join(path.Dir(executablePath), dbPath)
+	}
+	err := os.MkdirAll(dbPath[:len(dbPath)-len(dbName)], os.ModePerm)
+	if err != nil {
+		return nil, errors.Internal.Wrap(err, "failed to create directory")
+	}
+	dbConnection := sqlite.Open(dbPath)
+	db, err := gorm.Open(dbConnection, &gorm.Config{})
+	if err != nil {
+		return nil, errors.Internal.Wrap(err, "creating sqlite db failed")
+	}
+	return db, nil
+}
+
+func newGormDb(cfg domain.DbConfig) (*gorm.DB, error) {
+	switch cfg.Driver {
+	case "mysql":
+		return createMySQLDb(cfg.MySQL)
+	case "sqlite":
+		return createSQLiteDb(cfg.SQLite)
+	default:
+		return nil, errors.Internal.New("unsupported db driver")
```

Comment:
the error is misleading. Should print "no db driver specified" if the DB config is missing

Reply:
I think that in general everything is written correctly, but I can change this section to a more extended one.
```
case "": 
       return nil, errors.Internal.New("no db driver specified")
default:
       return nil, errors.Internal.Newf("unsupported db driver: %s" cfg.Driver)
```
@mikem-zed Will it fit?